tests/test_dmxrange_add_sub.py

```diff
@@ -0,0 +1,140 @@
+"""Test functions for adding and removing DMX ranges."""
+
+import pytest
+import io
+import numpy as np
+from pint.models import get_model
+from pint.models.dispersion_model import DispersionDM, DispersionDMX
+
+
+def test_startMJD_greaterThan_endMJD():
+    """ Check for error when start MJD is larger than end MJD. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58100, 58000, 3, 1, frozen=False)
+
+
+def test_only_one_MJD():
+    """ Check for error when one of the MJDs is None and the other isn't. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(None, 58000, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58000, None, 3, 1, frozen=False)
+
+
+def test_duplicate_index():
+    """ Check for error when a duplicate DMX index is used. """
+    dm_mod = DispersionDMX()
+    dm_mod.add_DMX_range(58000, 58100, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58200, 58300, 3, 1, frozen=False)
+
+
+def test_remove_nonexistent_index():
+    """ Check for error when a unused DMX index is removed. """
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.remove_DMX_range(3)
+
+
+def test_nonetype_MJD():
+    dm_mod = DispersionDMX()
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(None, 58000, 3, 1, frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod.add_DMX_range(58000, None, 3, 1, frozen=False)
+
+
+def test_dynamic_index():
+    dm_mod = DispersionDMX()
+    index1 = 3
```

Comment:
If you only use these variables once it's probably clearer to put the values in the right places directly (here and in other tests). It'll make them much shorter and easier to read.

Reply:
i thought it might be easier to be able to tell which variable are which in the call to add/remove the ranges. but i can change it if this doesn't make sense.